core/src/main/java/hudson/model/Executor.java

```diff
@@ -457,9 +457,9 @@ public SubTask call() throws Exception {
             LOGGER.log(SEVERE, getName()+": Unexpected executor death", e);
         } finally {
             if (asynchronousExecution == null) {
+            	executableEstimatedDuration = DEFAULT_ESTIMATED_DURATION;
```

Comment:
Some indenting issues, but LGTM in general. It would make sense to add a code comment explaining the behavior

Reply:
This line looks correct for me, but we need to drop cached value also at line 503. In `completedAsynchronous`.

Reply:
Yes, this method seems to be a better place for this modification. I adjusted the pull request accordingly.